src/coreclr/jit/emitxarch.cpp

```diff
@@ -12358,10 +12358,18 @@ void emitter::emitDispIns(
                 reg2          = reg3;
                 reg3          = tmp;
             }
+
+            emitAttr attr3 = attr;
+            if (hasTupleTypeInfo(ins) && ((insTupleTypeInfo(ins) & INS_TT_MEM128) != 0))
+            {
+                // Shift instructions take xmm for the 3rd operand regardless of instruction size.
+                attr3 = EA_16BYTE;
+            }
```

Comment:
Not quite sure this is the "best" fix.

This feels like a case where we could utilize the existing tags for memory operand size for the register as well rather than hardcoding a one-off special case.

Not going to block on it, especially since this is only used for disassembly; but its something that would be nice to have improved for this and other cases where the last operand is a different register size than the others.

Reply:
Can you elaborate on your preferred fix here?  This is already using the tuple type rather than hard-coded instruction list -- I just mention shift in the comment explicitly because it's the only thing I'm aware of that uses full xmm/m128 for what is ultimately an 8-bit scalar, and this should be the only encoding form where that xmm fix is required.